packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ImageAnalysisHostApiImpl.java

```diff
@@ -0,0 +1,125 @@
+
+  public ImageAnalysisHostApiImpl(
+      @NonNull BinaryMessenger binaryMessenger, @NonNull InstanceManager instanceManager) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+  }
+
+  /**
+   * Sets the context that will be used to run an {@link ImageAnalysis.Analyzer} on the main thread.
+   */
+  public void setContext(Context context) {
+    this.context = context;
+  }
+
+  /** Creates an {@link ImageAnalysis} instance with the target resolution if specified. */
+  @Override
+  public void create(@NonNull Long identifier, @Nullable ResolutionInfo targetResolution) {
+    ImageAnalysis.Builder imageAnalysisBuilder = cameraXProxy.createImageAnalysisBuilder();
+
+    if (targetResolution != null) {
+      imageAnalysisBuilder.setTargetResolution(CameraXProxy.sizeFromResolution(targetResolution));
+    }
+
+    ImageAnalysis imageAnalysis = imageAnalysisBuilder.build();
+    instanceManager.addDartCreatedInstance(imageAnalysis, identifier);
+  }
+
+  /**
+   * Sets an analyzer created by {@link ImageAnalysisHostApiImpl#createImageAnalysisAnalyzer()} on
+   * an {@link ImageAnalysis} instance to receive and analyze images.
+   */
+  @Override
+  public void setAnalyzer(@NonNull Long identifier) {
+    ImageAnalysis imageAnalysis =
+        (ImageAnalysis) Objects.requireNonNull(instanceManager.getInstance(identifier));
+    ImageAnalysis.Analyzer analyzer = createImageAnalysisAnalyzer();
+    imageAnalysis.setAnalyzer(ContextCompat.getMainExecutor(context), analyzer);
+  }
+
+  /**
+   * Creates an {@link ImageAnalysis.Analyzer} instance to send image information to the Dart side
+   * to support image streaming.
+   *
+   * <p>The image information collected and sent matches that of the (Dart) CameraImageData class,
+   * which is required for image streaming in this plugin.
+   */
+  private ImageAnalysis.Analyzer createImageAnalysisAnalyzer() {
+    return new ImageAnalysis.Analyzer() {
+      @Override
+      public void analyze(@NonNull ImageProxy image) {
+        // Collect image plane information.
+        ImageProxy.PlaneProxy[] planes = image.getPlanes();
+        List<GeneratedCameraXLibrary.ImagePlaneInformation> imagePlanesInformation =
+            new ArrayList<GeneratedCameraXLibrary.ImagePlaneInformation>();
+
+        for (ImageProxy.PlaneProxy plane : planes) {
+          ByteBuffer byteBuffer = plane.getBuffer();
+          byte[] bytes = new byte[byteBuffer.remaining()];
+          byteBuffer.get(bytes, 0, bytes.length);
+          GeneratedCameraXLibrary.ImagePlaneInformation.Builder imagePlaneInfoBuilder =
+              new GeneratedCameraXLibrary.ImagePlaneInformation.Builder();
+
+          imagePlanesInformation.add(
+              imagePlaneInfoBuilder
+                  .setBytesPerRow(Long.valueOf(plane.getRowStride()))
+                  .setBytesPerPixel(Long.valueOf(plane.getPixelStride()))
+                  .setBytes(bytes)
```

Comment:
Have you considered passing back an instance of `ImageProxy` and separating the calls to `getPlanes`. As in,

```dart
class ImageProxy {
  Future<List<ImageAnalysisPlane>> getPlanes() {
    // ... 
  }
}

class ImageProxyPlane {
}
``` 

In the long run, I think we would not want to pass back the bytes for every frame by default. Many use cases of ImageAnalysis won't be able to run analysis on every frame and this could be detrimental for performance.